File: .cursor/tools/cleanup_temp_backups.py
#!/usr/bin/env python3
from __future__ import annotations

import argparse
import shutil
from datetime import datetime, timedelta
from pathlib import Path


def is_temp_file(file_path: Path) -> bool:
    if file_path.name.startswith("TMP_"):
        return True
    try:
        text = file_path.read_text(encoding="utf-8", errors="ignore")
    except OSError:
        return False
    head = "\n".join(text.splitlines()[:30])
    return "Lifecycle: 临时" in head


def move_soft_delete(src: Path, trash_root: Path) -> Path:
    rel = src.name
    target = trash_root / rel
    if target.exists():
        stamp = datetime.now().strftime("%Y%m%d%H%M%S")
        target = trash_root / f"{src.stem}_{stamp}{src.suffix}"
    target.parent.mkdir(parents=True, exist_ok=True)
    shutil.move(str(src), str(target))
    return target


def main() -> int:
    parser = argparse.ArgumentParser()
    parser.add_argument("--days", type=int, default=15)
    args = parser.parse_args()

    root = Path(__file__).resolve().parents[2]
    daily_dir = root / "Daily-Backups"
    trash_dir = root / ".trash" / "soft-delete" / "daily-backups"
    cutoff = datetime.now() - timedelta(days=args.days)

    if not daily_dir.exists():
        print("No Daily-Backups directory found.")
        return 0

    moved = []
    for file_path in daily_dir.glob("*.md"):
        if file_path.name == "日常备份索引.md":
            continue
        if not is_temp_file(file_path):
            continue
        mtime = datetime.fromtimestamp(file_path.stat().st_mtime)
        if mtime <= cutoff:
            target = move_soft_delete(file_path, trash_dir)
            moved.append((file_path.name, str(target)))

    print(f"Scanned: {daily_dir}")
    print(f"Retention days: {args.days}")
    print(f"Moved count: {len(moved)}")
    for name, target in moved:
        print(f"- {name} -> {target}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: adapters/cursor/tools/cleanup_temp_backups.mjs
import fs from "node:fs";
import path from "node:path";

function readHead(filePath, lines = 30) {
  const text = fs.readFileSync(filePath, "utf8");
  return text.split(/\r?\n/).slice(0, lines).join("\n");
}

function isTempFile(filePath) {
  const base = path.basename(filePath);
  if (base.startsWith("TMP_")) return true;
  const head = readHead(filePath);
  return head.includes("Lifecycle: 临时");
}

function moveSoftDelete(src, trashRoot) {
  const ext = path.extname(src);
  const stem = path.basename(src, ext);
  let target = path.join(trashRoot, path.basename(src));
  if (fs.existsSync(target)) {
    const stamp = new Date().toISOString().replace(/[-:TZ.]/g, "").slice(0, 14);
    target = path.join(trashRoot, `${stem}_${stamp}${ext}`);
  }
  fs.mkdirSync(path.dirname(target), { recursive: true });
  fs.renameSync(src, target);
  return target;
}

const argDays = process.argv.find((a) => a.startsWith("--days="));
const days = argDays ? Number(argDays.split("=")[1]) : 15;
const now = Date.now();
const cutoff = now - days * 24 * 60 * 60 * 1000;

const root = path.resolve(path.dirname(new URL(import.meta.url).pathname), "..", "..").replace(/^\/([A-Za-z]:)/, "$1");
const dailyDir = path.join(root, "Daily-Backups");
const trashDir = path.join(root, ".trash", "soft-delete", "daily-backups");

if (!fs.existsSync(dailyDir)) {
  console.log("No Daily-Backups directory found.");
  process.exit(0);
}

const moved = [];
for (const name of fs.readdirSync(dailyDir)) {
  if (!name.endsWith(".md")) continue;
  if (name === "日常备份索引.md") continue;
  const full = path.join(dailyDir, name);
  if (!isTempFile(full)) continue;
  const stat = fs.statSync(full);
  if (stat.mtimeMs <= cutoff) {
    const target = moveSoftDelete(full, trashDir);
    moved.push({ name, target });
  }
}

console.log(`Scanned: ${dailyDir}`);
console.log(`Retention days: ${days}`);
console.log(`Moved count: ${moved.length}`);
for (const item of moved) {
  console.log(`- ${item.name} -> ${item.target}`);
}
